twoliter/embedded/Makefile.toml

```diff
@@ -829,6 +856,17 @@ export BUILDSYS_VARIANT_RUNTIME="${BUILDSYS_VARIANT_RUNTIME:?}"
 export BUILDSYS_VARIANT_FAMILY="${BUILDSYS_VARIANT_FAMILY:?}"
 export BUILDSYS_VARIANT_FLAVOR="${BUILDSYS_VARIANT_FLAVOR}"
 
+# Relocate an existing target directory to the new location, to avoid breaking
+# compatibility on upgrade. On downgrade, the missing directory will trigger a
+# full rebuild, which is good because the new layout of RPM artifacts is not
+# compatible with previous versions.
+if [ -d "${BUILDSYS_ROOT_DIR}/variants/target/${BUILDSYS_ARCH}" ]; then
+  mkdir -p "${BUILDSYS_ROOT_DIR}/target"
+  mv \
```

Comment:
Don't you also need to check if `"${BUILDSYS_ROOT_DIR}/Cargo.toml` exists?

Reply:
> Don't you also need to check if "${BUILDSYS_ROOT_DIR}/Cargo.toml exists?

This works regardless of where the manifest workspace is located - it just relocates the targets output directory. That helps with downgrade compatibility since then `cargo` won't find it with an older `twoliter` release and will rebuild everything.